letsencrypt/models/base_config_settings.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+from inspect import getmembers, isfunction
+
+
+class BaseConfigSettings(models.TransientModel):
+    _inherit = 'base.config.settings'
+
+    letsencrypt_dns_provider = fields.Selection(
+        selection='_get_supported_dns_providers')
+    letsencrypt_dns_login = fields.Char()
+    letsencrypt_dns_key = fields.Char()
+    letsencrypt_dns_url = fields.Char()
+    letsencrypt_needs_dns_provider = fields.Boolean()
+
+    def _get_supported_dns_providers(self):
+        providers = []
+        for func in [x for x in getmembers(
+            self, lambda x: isfunction(x) and
+                x.__name__.startswith('letsencrypt_') and
+                x.__name__.endswith('_support'))]:
+            provider = x[12:-8]
+            providers.append((provider, provider))
+        return providers
+
+    @api.multi
+    def default_get(self):
+        self.ensure_one()
+        res = super(BaseConfigSettings, self).default_get()
+        ir_config_parameter = self.env['ir.config_parameter']
+        res.update({
+            'letsencrypt_dns_provider': ir_config_parameter.get_param(
+                'letsencrypt_dns_provider'),
+            'letsencrypt_dns_login': ir_config_parameter.get_param(
+                'letsencrypt_dns_login'),
+            'letsencrypt_dns_key': ir_config_parameter.get_param(
+                'letsencrypt_dns_key'),
+            'letsencrypt_dns_url': ir_config_parameter.get_param(
+                'letsencrypt_dns_url'),
+            'letsencrypt_needs_dns_provider': ir_config_parameter.get_param(
+                )
+        })
+        return res
+
+    @api.multi
+    def set_dns_provider(self):
+        self.ensure_one()
+        ir_config_parameter = self.env['ir.config_parameter']
```

Comment:
this should call some `set_letencrypt_dns_$provider` so that every override can add parameters as necessary and save them in a way that's convenient

Reply:
As I am aware, in a `res.config.settings` model the `set*` functions are automatically called on save. Why would I call a function by myself? Wouldn't it make sense if they define a `set_*` themselves in their module?

Reply:
yes, that's better